### ISSUE TITLE
Does not work Minecraft 1.16.1 with LWJGW 3.3.2

### ISSUE THREAD

**Issue by naari3** [FabricMC/fabric-loom#1170]
I downloaded the 1.16.1 [template](https://fabricmc.net/develop/template/) and tried to launch the client from the launch configuration generated by `.\gradlew vscode`, but it crashed with the following log:

https://mclo.gs/T3fg5gE

What I found interesting here was that the LWJGL version was 3.3.2 even though the log was for 1.16.1.

If I understand correctly, Minecraft 1.16.1 uses LWJGL 3.2.1 or 3.2.2, and the error in this log seems to be caused by a difference in the version of LWJGL.

ref: https://piston-meta.mojang.com/v1/packages/54fa3af57d041d2771e66d390197b2c0288e697c/1.16.1.json

I created an issue because I don't know how the LWJGL-related libraries under `.gradle/loom-cache/natives/1.16.1/windows` are downloaded and deployed here, and I would like to know if there is a way to change this version.

**Comment by modmuss50** [comment]
Loom upgrades LWJGL when required, e.g to support a newer version of Java or to support a platform such as ARM64 MacOS that was previously unsupported.

The rules for this are located here: https://github.com/FabricMC/fabric-loom/blob/dev/1.7/src/main/java/net/fabricmc/loom/configuration/providers/minecraft/library/processors/LWJGL3UpgradeLibraryProcessor.java#L53-L60

Disabling it would just lead to a crash at runtime anyway. You still compile against the older version so your mod should remain compatible.

The fact that its causing a crash becuase it was upgraded isnt ideal at all, and something that I should look into. Maybe try using Java 17?

**Comment by naari3** [comment]
> The fact that its causing a crash becuase it was upgraded isnt ideal at all, and something that I should look into. Maybe try using Java 17?

Does this mean that 1.16.1 with LWJGW 3.3.2 should work? I'm not familiar with interacting with LWJGL and would love to hear about this. thanks.

**Comment by modmuss50** [comment]
Yes, I expected this to work, and it does for most versions of the game. I think this version went untested and this issue wasnt noticed. I am confident I can find a solution that will enable you to use Java 21+ on 1.16.1, for now just stick to Java 17.

**Comment by naari3** [comment]
Just to be clear, I tested it with Java 17 and it worked fine (works on LWJGW 3.2.2).

**Comment by naari3** [comment]
I changed the title, but if you find something better please change it to that. 🙏

**Comment by modmuss50** [comment]
Hi, I am unable to reproduce this crash, im using Java 21 and Minecraft 1.16.1, can you post the full startup log including mod list please?

**Comment by naari3** [comment]
Here it is. Is this enough? If there is anything missing, please let me know and I will take care of it.
https://mclo.gs/qBUnDFS

**Comment by modmuss50** [comment]
Yes, thanks thats helpful. I see you are on Windows I wonder if that has something to do with it, I will test this myself on Windows.

**Comment by modmuss50** [comment]
On windows it still seems to work for me, at least when I run `gradlew.bat runClient`, and via intelij. I wonder if this is a VSCode issue?

Could you possibly try this command to see if it works, if not `gradlew.bat runClient --refresh-dependencies`.

**Comment by naari3** [comment]
Oh sorry, I forgot to mention in the issue that it was the [VSCode Launch Target](https://fabricmc.net/wiki/tutorial:setup:vscode#generating_run_configurations) that failed this time!

I ran `runClient` task from the console and it started without any issues! so this looks like a launch.json or vscode issue?.

and, I also tested launch target `Minecraft Client` with a clean profile with only the extensions we needed installed but it still doesn't work. (this is not a exactly clean environment, but It takes too much time to setup a clean vscode on my computer sorry)
